acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -5,22 +5,34 @@
 require 'openssl'
 require 'socket'
 require 'json'
+require 'securerandom'
 
 # This file contains general test helper methods for pxp-agent acceptance tests
 
 # The standard path for git checkouts is where pcp-broker will be
 GIT_CLONE_FOLDER = '/opt/puppet-git-repos'
 
 # Some helpers for working with the log file
-PXP_LOG_FILE_CYGPATH = '/cygdrive/c/ProgramData/PuppetLabs/pxp-agent/var/log/pxp-agent.log'
-PXP_LOG_FILE_POSIX = '/var/log/puppetlabs/pxp-agent/pxp-agent.log'
+PXP_LOG_DIR_CYGPATH = '/cygdrive/c/ProgramData/PuppetLabs/pxp-agent/var/log'
+PXP_LOG_DIR_POSIX = '/var/log/puppetlabs/pxp-agent'
+
+def logdir(host)
+  windows?(host)?
+    PXP_LOG_DIR_CYGPATH :
+    PXP_LOG_DIR_POSIX
+end
 
 # @param host the beaker host you want the path to the log file on
 # @return The path to the log file on the host. For Windows, a cygpath is returned
 def logfile(host)
-  windows?(host)?
-    PXP_LOG_FILE_CYGPATH :
-    PXP_LOG_FILE_POSIX
+  logdir(host)+'/pxp-agent.log'
+end
+
+def reset_logfile(host)
+  old_logfile = logfile(host)
+  new_logfile = old_logfile+'.'+SecureRandom.uuid
+  logger.notify "---- Saving pxp-agent logfile to #{new_logfile}"
+  retry_on(host, "test -f #{old_logfile} && mv #{old_logfile} #{new_logfile}")
```

Comment:
This still has the issue of erroring due to a non-zero exit code in cases where the log file doesn't exist.
It could be made safe for the case of the log file not existing by inverting the test like:

    retry_on(host, "test ! -f #{old_logfile} || mv #{old_logfile} #{new_logfile}")